skillmodels/pre_processing/data_processor.py

```diff
@@ -275,3 +294,20 @@ def reg_df(self, factor, period=None, stage=None, controls=None, agg_method="mea
 
         reg_df = pd.concat(period_dfs, axis=0)
         return reg_df
+
+    def _check_observable_data(self):
+        report_issues = pd.DataFrame()
+        for t in self.periods:
+            df = self.data.query(f"__period__ == {t}")
+            # create time relevant measurements list
+            meas_t = []
+            for fac in self.factors:
```

Comment:
If you use `meas_t += self.measurements[fac][t]` you don't have to flatten afterwards.